### PR TITLE
​

### PR DESCRIPTION
**Pull request by 74togo**

### CODE REVIEW

File: src/chip8.rs [hunk 1]
```diff
@@ -10,6 +10,7 @@ pub struct Chip8 {
     pub st: u8,
     pc: u16,
     display: [[u8; 64]; 32],
+    pub display_changed: bool,
```
Comment (PhearTheCeal) [inline src/chip8.rs]: fine grained display control

File: src/chip8.rs [hunk 2]
```diff
@@ -224,89 +226,101 @@ impl Chip8 {
...
     // 7xkk - ADD Vx, byte
     // Set Vx = Vx + kk.
     fn add_value_to_reg(&mut self, register: usize, value: u8) {
-        let x = self.registers[register];
-        let kk = value as u8;
+        let x = self.registers[register] as u16;
+        let kk = value as u16;
 
-        let (result, overflow) = x.overflowing_add(kk);
-        self.registers[0xf] = overflow as u8;
```
Comment (PhearTheCeal) [inline src/chip8.rs]: VF shouldn't be affected by this op

File: src/chip8.rs [hunk 3]
```diff
@@ -107,15 +109,12 @@ impl Chip8 {
...
+        self.display_changed = false;
         // combine two bytes to get the full opcode, as a u16
         let opcode = (self.memory[self.pc as usize] as u16) << 8
             | self.memory[(self.pc + 1) as usize] as u16;
 
         self.execute_opcode(opcode);
-
-        // We step forward after each instruction, which is why you'll see self.pc -= 2
-        // in a couple places, like jumping and returning.
-        self.pc += 2;
```
Comment (PhearTheCeal) [inline src/chip8.rs]: don't like this, implementing the spec instead

File: src/chip8.rs [hunk 4]
```diff
@@ -315,53 +329,49 @@ impl Chip8 {
...
+        for byte in 0..(n as usize) {
+            let y = (self.registers[y] as usize + byte) % 32;
+            for bit in 0..8 {
+                let x = (self.registers[x] as usize + bit) % 64;
+                let color = (self.memory[self.i as usize + byte] >> (7 - bit)) & 1;
+                self.registers[0xf] |= color & self.display[y][x];
+                self.display[y][x] ^= color;
             }
         }
+        if self.registers[0xf] == 0 {
```
Comment (PhearTheCeal) [inline src/chip8.rs]: forum nerds hate him

File: src/chip8.rs [hunk 5]
```diff
@@ -431,32 +445,27 @@ impl Chip8 {
...
+        self.memory[i + 1] = (n / 10)  % 10;
+        self.memory[i + 2] = (n / 1)   % 10;
+        self.pc += 2;
     }
 
     // Fx55 - LD [I], Vx
     // Store registers V0 through Vx in memory starting at location I.
     fn copy_registers_into_memory(&mut self, max_register: usize) {
-        for x in 0..=max_register {
+        for x in 0..max_register + 1 {
```
Comment (kingcritter) [inline src/chip8.rs]: 🤔

File: src/main.rs [hunk 6]
```diff
@@ -58,7 +56,7 @@ fn main() {
     let video_subsystem = sdl_context.video().unwrap();
     let window = video_subsystem
         .window(
-            "Critter's Amazing Chip8 Emulator",
+            "PTC's Working Chip8 Emulator",
```
Comment (kingcritter) [inline src/main.rs]: I believe that in the parlance of Twitter, this is called "throwing shade."

**Review by kingcritter — COMMENTED** [review]
悪くないよ

File: src/main.rs (hunk 6 above)
Comment (PhearTheCeal) [reply src/main.rs]: forgot about that lol, was my first change just to see how rust compiles/runs stuff

File: src/chip8.rs (hunk 5 above)
Comment (PhearTheCeal) [reply src/chip8.rs]: both are ugly but at least one is a widely accepted ugly since the 70s

**Pull request closed without merging**